doc/source/structures/misc/gui.rst

```diff
@@ -0,0 +1,304 @@
+    ===================================== =============================== =============
+
+.. structure:: Widget
+
+    This object is the base class of all GUI elements.
+
+    ===================================== =============================== =============
+    Suffix                                Type                            Description
+    ===================================== =============================== =============
+    :meth:`SHOW`                                                          Show the widget. All except GUI objects are shown by default.
+    :meth:`HIDE`                                                          Hide the widget.
+    :meth:`DISPOSE`                                                       Remove the widget permanently.
+    :attr:`ENABLED`                       :struct:`Boolean`               Set to False to "grey out" the widget, preventing user interaction.
+    :attr:`HMARGIN`                       :struct:`scalar` (pixels)       Horizontal spacing between this and other widgets.
+    :attr:`VMARGIN`                       :struct:`scalar` (pixels)       Vertical spacing between this and other widgets.
+    :attr:`HPADDING`                      :struct:`scalar` (pixels)       Horizontal spacing between the outside of the widget and its contents.
+    :attr:`VPADDING`                      :struct:`scalar` (pixels)       Vertical spacing between the outside of the widget and its contents.
+    :attr:`HSTRETCH`                      :struct:`Boolean`               Should the widget stretch horizontally? (default depends on widget subclass)
+    :attr:`VSTRETCH`                      :struct:`Boolean`               Should the widget stretch vertically?
+    :attr:`WIDTH`                         :struct:`scalar` (pixels)       Fixed width (or 0 if flexible).
+    :attr:`HEIGHT`                        :struct:`scalar` (pixels)       Fixed height (or 0 if flexible).
+    :attr:`BG`                            :struct:`string`                Name of a "9-slice" image file. See note below.
+    :attr:`HBORDER`                       :struct:`scalar` (pixels)       Left and right column counts for BG image border.
+    :attr:`VBORDER`                       :struct:`scalar` (pixels)       Top and bottom row counts for BG image border.
+    ===================================== =============================== =============
+
+.. note::
+
+    The `BG` attribute comes in 8 flavours: `BG`, `BG_FOCUSED`, `BG_ACTIVE`, `BG_ON`, `BG_FOCUSED_ON`, `BG_ACTIVE_ON`, and `BG_HOVER_ON`.
+
+    ============= =====================================================
+    FOCUSED       The widget has keyboard focus.
+    ACTIVE        The widget is hot (eg. button is being held down).
+    HOVER         The widget is under the mouse.
+    ON            The widget is on (eg. button is PRESSED).
+    ============= =====================================================
+
```

Comment:
They're in src/kOS/Suffixed/Widgets.cs in the original PR (src/kOS/Suffixed/Widget/Button.cs now) - note that github elides it because it's a long file, so CTRL-F won't find it.